Mathlib/Algebra/Lie/Weights/Cartan.lean

```diff
@@ -276,33 +246,51 @@ theorem rootSpace_zero_eq (H : LieSubalgebra R L) [H.IsCartanSubalgebra] [IsNoet
 variable {R L H}
 variable [H.IsCartanSubalgebra] [IsNoetherian R L] (α : H → R)
 
-/-- Given a root `α`, the Lie bracket restricted to the product of the root space of `α` and `-α`
-takes value in the Cartan subalgebra.
+/-- Given a root `α` relative to a Cartan subalgebra `H`, this is the span of all products of
+an element of the `α` root space and an element of the `-α` root space. Informally it is often
+denoted `⁅H(α), H(-α)⁆`.
+
+When `L` is semisimple over a field of characteristic zero, it is spanned by corresponding coroot
+corresponding to `α`, see `LieAlgebra.IsKilling.coe_corootSpace_eq_span_singleton`.
 
-When `L` is semisimple, the image of this map is one-dimensional and is spanned by the corresponding
-coroot. -/
-def rootSpaceProductNegSelf : rootSpace H α ⊗[R] rootSpace H (-α) →ₗ⁅R,H⁆ H :=
-  ((rootSpace H 0).incl.comp <| rootSpaceProduct R L H α (-α) 0 (add_neg_self α)).codRestrict
-    H.toLieSubmodule (by
+Note that the name "coroot space" is not standard as this space does not seem to have a name in the
+informal literature. -/
+def corootSpace : LieIdeal R H :=
+  LieModuleHom.range <| ((rootSpace H 0).incl.comp <|
+    rootSpaceProduct R L H α (-α) 0 (add_neg_self α)).codRestrict H.toLieSubmodule (by
   rw [← rootSpace_zero_eq]
   exact fun p ↦ (rootSpaceProduct R L H α (-α) 0 (add_neg_self α) p).property)
 
-@[simp]
-lemma coe_rootSpaceProductNegSelf_apply (x : rootSpace H α) (y : rootSpace H (-α)) :
-    (rootSpaceProductNegSelf α (x ⊗ₜ y) : L) = ⁅(x : L), (y : L)⁆ :=
-  rfl
-
-lemma mem_range_rootSpaceProductNegSelf {x : H} :
-    x ∈ (rootSpaceProductNegSelf α).range ↔
+lemma mem_corootSpace {x : H} :
+    x ∈ corootSpace α ↔
     (x : L) ∈ Submodule.span R {⁅y, z⁆ | (y ∈ rootSpace H α) (z ∈ rootSpace H (-α))} := by
-  have : x ∈ (rootSpaceProductNegSelf α).range ↔
-      (x : L) ∈ (rootSpaceProductNegSelf α).range.map H.toLieSubmodule.incl := by
+  have : x ∈ corootSpace α ↔
+      (x : L) ∈ LieSubmodule.map H.toLieSubmodule.incl (corootSpace α) := by
+    rw [corootSpace]
     simpa using exists_congr fun _ ↦ H.toLieSubmodule.injective_incl.eq_iff.symm
-  simp_rw [this, ← LieModuleHom.map_top, ← LieSubmodule.mem_coeSubmodule,
+  simp_rw [this, corootSpace, ← LieModuleHom.map_top, ← LieSubmodule.mem_coeSubmodule,
     LieSubmodule.coeSubmodule_map, LieSubmodule.top_coeSubmodule, ← TensorProduct.span_tmul_eq_top,
     LinearMap.map_span, Set.image, Set.mem_setOf_eq, exists_exists_exists_and_eq]
   change (x : L) ∈ Submodule.span R
     {x | ∃ (a : rootSpace H α) (b : rootSpace H (-α)), ⁅(a : L), (b : L)⁆ = x} ↔ _
   simp
 
+lemma mem_corootSpace' {x : H} :
+    x ∈ corootSpace α ↔
+    x ∈ Submodule.span R {u : H | ∃ᵉ (y ∈ rootSpace H α) (z ∈ rootSpace H (-α)), ⁅y, z⁆ = u} := by
```

Comment:
Does this work? Also on the line below?
```suggestion
    x ∈ Submodule.span R {⁅y, z⁆ : H | (y ∈ rootSpace H α) (z ∈ rootSpace H (-α))} := by
```

Reply:
Not quite but a slight variation does, thanks:
```suggestion
    x ∈ Submodule.span R ({⁅y, z⁆ | (y ∈ rootSpace H α) (z ∈ rootSpace H (-α))} : Set H) := by
```